레스토랑 상세 페이지 구현하기

hanvit7 opened the pull request:
안녕하세요 트레이너님 
바쁘다는 핑계로 이제서야 리뷰요청드립니다.
이제 시작해서 별로 진행은 못했습니다. ㅠㅜ

hanvit7 committed:
react-router-dom 설치

hanvit7 committed:
webpack historyApiFallback 설정

hanvit7 committed:
리팩토링 - RestaurantsPage 생성

- 기존 App에 있는 내용을 RestaurantsPage로 추출

hanvit7 committed:
구현 - App에 window객채를 이용한 라우터 구현

- HomePage, NotFoundPage, AboutPage를 함수로 구현
- window객채를 이용해 각각의 페이지로 라우팅 구현

hanvit7 committed:
테스트 - app 테스트 에러 수정

hanvit7 committed:
구현 - HomePage 추출

hanvit7 committed:
구현 - AboutPage 추출

hanvit7 committed:
구현 - NotFoundPage 추출

hanvit7 committed:
구현 - App에 React Router 적용

- 기존 라우팅 구현을 대체하여 BrowserRouter 적용

hanvit7 committed:
구현 - Home에 링크를 <a>에서 <Link>로 변경

moonkii commented:
![image](https://user-images.githubusercontent.com/46776329/121472165-c5f3de00-c9fb-11eb-91fc-841eed97bcc5.png)

간단한 페이지 하나를 만들더라도 TDD 싸이클을 잘 지켜가면서 진행해주세요!

hanvit7 committed:
테스트 - HomePage 테스트 오류 수정

- MemoryRouter 추가

hanvit7 committed:
리팩토링 - route 테스트를 위해 BrowserRouter 위치 이동

- app에서 index로 이동

hanvit7 committed:
테스트 - App 테스트 추가

- HomePage 랜터링 확인
- AboutPage 랜터링 확인

hanvit7 committed:
리팩토링 - renderApp 추출

- app render시 MemoryRouter에 path 매개변수 받는 함수 추출

hanvit7 committed:
테스트 - App 테스트 추가

- Restaurants Page 랜터링 확인
- Not Found Page 랜터링 확인

hanvit7 committed:
테스트 - RestaurantsContainer 테스트 추가

- restaurants/{id} 링크를 가지고 있는지 테스트

hanvit7 committed:
구현 - Restaurants 상세페이지 링크 추가

- restaurants/:id 주소로 레스토랑 제목에 링크 추가

hanvit7 committed:
테스트 - RestaurantsPage Link 추가로 테스트 깨짐 수정

hanvit7 committed:
구현 - App에 restaurants/:id 라우팅 추가

hanvit7 committed:
테스트 - RestaurantDetailPage 테스트 추가

hanvit7 committed:
구현 - RestaurantDetailPage 기본틀 구현

hanvit7 committed:
테스트 - RestaurantDetailPage 테스트 추가

- useParams 값 테스트 추가

hanvit7 commented:
안녕하세요 트레이너님 오랜만에....PR입니다.
시간을 못쓰기도 한데다 useParam을 테스트 하는데에서 헤매다 이제왔습니다ㅠㅜ

moonkii commented on src/RestaurantsPage.test.jsx:
```diff
@@ -0,0 +1,38 @@
+
+import RestaurantsPage from './RestaurantsPage';
+
+test('RestaurantsPage', () => {
```
웬만한 경우 describe ~ (context) ~ it 형태로 테스트를 작성해주시면
테스트를 작성하는 의도가 시나리오에서 드러나기 때문에 가독성이 더 좋아요 ;)

hanvit7 committed:
구현 - RestaurantDetailPage에서 loadRestaurant 구현

hanvit7 committed:
테스트 - RestaurantDetailPage 테스트 수정

- dispatch 추가

hanvit7 committed:
구현 - loadRestaurant 구현

action
- loadRestaurant, setRestaurant action 구현
api
- fetchRestaurant 구현
- fetchRestaurant mock 추가

hanvit7 committed:
테스트 - reducer.test에 setRestaurant 테스트 추가

hanvit7 committed:
구현 - reducer에 setRestaurant 구현

hanvit7 committed:
테스트 - RestaurantDetailPage 테스트 추가

- render loading 테스트 추가
- load dispatch 테스트 추가
- render restaurant name 테스트 추가

hanvit7 committed:
구현 - RestaurantDetailPage 구현

- restaurant 로딩 표시
- restaurant detail 정보 표시

moonkii commented:
과제를 잘 수행하셨어요! 다음주 부터는 PR을 미리 보내서 더 많은 리뷰를 받으시면 좋을 것 같아요 ;)
회고 작성 후 공유해 주시면 merge 하도록하겠습니다!
